22535 faces mbean

tbitonti opened the pull request:
For issue #22535: Update the faces schema to the 4.0 final version, and update the parser to match.
That means, faces parsing no longer recognizes managed-bean elements when the schema version is 4.0 or higher.

tbitonti commented:
Tests added:
- com.ibm.ws.javaee.ddmodel.jsf.JSFAppTest
  - testJSF30_Managedbean()
    - Verify parse of "managed-bean" element when the schema version is 3.0
  - testJSF40_Managedbean
    - Verify that "managed-bean" is not handled when the schema version is 4.0

tbitonti commented:
#build

tbitonti commented:
Pretty much complete build failure.  The WAS Liberty side of the build seems to have failed completely.  Retrying.

tbitonti commented:
#build

pnicolucci commented on dev/com.ibm.ws.javaee.dd/src/com/ibm/ws/javaee/dd/web/common/package-info.java:
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2010 IBM Corporation and others.
+ * Copyright (c) 2010,2022 IBM Corporation and others.
```
Minor detail but usually there is a space after `,` in the date range.

tbitonti replied:
Fixed.

pnicolucci commented on dev/com.ibm.ws.javaee.ddmodel/test/com/ibm/ws/javaee/ddmodel/jsf/JSFAppTest.java:
```diff
@@ -349,4 +352,38 @@ public void testJSF23_SearchKeywordResolver() throws Exception {
+    // The 'managed-bean' element should be parsed at faces 3.0:
+    
+    @Test
+    public void testJSF30_Managedbean() throws Exception {    
```
We should try out best to start referring to `JSF` as `Faces` going forward. Would you mind updating the test names from JSF30/JSF40 to Face30/Faces40?

tbitonti replied:
Fixed

tbitonti commented:
One unrelated test failure: The test timed out:

```
- com.ibm.ws.security.saml.sso_fat.2 FAT tests
  - com.ibm.ws.security.saml.fat.FATSuite
    - userFeatureSAMLTests_mainFlowTest_EE9_FEATURES

```

tbitonti committed:
Update web-facesconfig_4_0.xsd to final version.
Remove "managed-bean" handling when the faces config schema version is 4.0 or higher.
Add "managed-bean" unit tests for faces 3.0 and faces 4.0.
Corrected copyright format.
Changed new Faces test names to use prefix "testFaces" instead of "testJSF".

tbitonti commented:
#build

tbitonti commented:
#libby

pnicolucci commented:
@tbitonti what is left to get this merged in?

pnicolucci commented:
#libby #build

tbitonti commented:
#build
